pkg/instrumentation/helper.go

```diff
@@ -133,3 +129,32 @@ func volumeSize(quantity *resource.Quantity) *resource.Quantity {
 	}
 	return quantity
 }
+
+func isValidContainersAnnotation(containersAnnotation string) error {
+	if containersAnnotation == "" {
+		return nil
+	}
+
+	matched, err := regexp.MatchString("^[a-zA-Z0-9-,]+$", containersAnnotation)
+	if err != nil {
+		return fmt.Errorf("error while checking for instrumentation container annotations %w", err)
+	}
+	if !matched {
+		return fmt.Errorf("not valid characters included in the instrumentation container annotation %s", containersAnnotation)
+	}
+	return nil
+}
+
+func configureLanguageContainers(inst *instrumentationWithContainers, annotation string, ns metav1.ObjectMeta, pod metav1.ObjectMeta) error {
```

Comment:
I think a better name for this would be `setContainersFromAnnotation`. We should also add a docstring here imo.